internals/scripts/extract-intl.js

```diff
@@ -14,14 +14,14 @@ const pkg = require('../../package.json');
 const presets = pkg.babel.presets;
 const plugins = pkg.babel.plugins || [];
 
-const i18n = require('../../app/i18n');
-import { DEFAULT_LOCALE } from '../../app/containers/App/constants';
-
 require('shelljs/global');
 
+import { DEFAULT_LOCALE } from '../../app/containers/App/constants';
+import { appLocales } from '../../app/i18n';
```

Comment:
@mxstbr : `import { DEFAULT_LOCALE } from '../../app/containers/App/constants';` was already there in master branch hence I just followed.  

`npm run extract-intl` => ` babel-node --presets latest,stage-0 -- ./internals/scripts/extract-intl.js` 

Hence `import` very well works.  

However `node internals/scripts/extract-intl.js` will not work as you said. 
https://github.com/p10ns11y/react-boilerplate/blob/d109f617f57eaae5fc7a08f7c91ad40477ab29be/internals/scripts/extract-intl.js#L100

Using arrow `async` function which is not available in node as well. Then we need to rewrite the entire script native node.